service/textmagic/textmagic.go

```diff
@@ -0,0 +1,54 @@
+package textmagic
+
+import (
+	"context"
+	tm "github.com/textmagic/textmagic-rest-go-v2/v2"
+	"strings"
+)
+
+//TMagicService allow you to configure a TextMagic SDK client.
+type TMagicService struct {
+	UserName           string
+	APIKey             string
+	destinations       []string
+	TextMagicAPIClient *tm.APIClient
+}
+
+//NewTextMagicClient creates a new text magic client
+func NewTextMagicClient(userName, apiKey string) *TMagicService {
+
+	config := tm.NewConfiguration()
+	client := tm.NewAPIClient(config)
+
+	return &TMagicService{
+		TextMagicAPIClient: client,
+		UserName:           userName,
+		APIKey:             apiKey,
+	}
+}
```

Comment:
```suggestion
// New creates a new text magic client. Use your user-name and API key from
// https://my.textmagic.com/online/api/rest-api/keys.
func New(userName, apiKey string) *Service {
	config := textMagic.NewConfiguration()
	client := textMagic.NewAPIClient(config)

	return &Service{
		client: client,
		userName:           userName,
		apiKey:             apiKey,
	}
}
```

Reply:
Function should just be called `New`; `textmagic.New()` is less redundant than `textmagic.NewTextMagicClient()`. Also, applying the new field names and moving comment about username and api key from `Send` method to here.